### ISSUE TITLE
Sensor data fusion if using more then one sensor

### ISSUE THREAD

**Issue by Cinezaster** [mdsiraj1992/Gassensors#4]
Some of the MQ sensors sense same gasses. 
If combining these data we could merge data of particular gasses
for example:
- MQ-2: LPG, Propane, H2, Methane, CH4, CO
- MQ-4: CH4, LPG, H2, CO
- MQ-9: CH4, CO, Methane
We could combine these for beter output of CH4, LPG, H2, CO, ...

**Comment by klynkins** [comment]
Hi Cinezaster,
did you ever combine those sensors to get a better reading. I am testing MQ2-MQ9 but feel we do not need to use all those sensors especially if they are repetitive gases being measured. I am curious if you have tested all together and if you got a better output?

Katelynn

**Comment by louxiantuo** [comment]
yeah,I only tested MQ9 sensor in arduino UNO.And I don't know what will happen if you use those sensors all together.I used code in github if you need them,I can give you url.

发自网易邮箱大师

在2017年08月23日 23:59，Katelynn Campbell<mailto:notifications@github.com> 写道：

Hi Cinezaster,
did you ever combine those sensors to get a better reading. I am testing MQ2-MQ9 but feel we do not need to use all those sensors especially if they are repetitive gases being measured. I am curious if you have tested all together and if you got a better output?

Katelynn

**Comment by Cinezaster** [comment]
@klynkins 
Yes but it is a complete different process.
We only measure raw data and measure against a reference gas, to calibrate these sensors and do post processing on the raw data and use the data with the reference gas as a baseline to measure against.
The methode used with this code is not representative although it's a clever trick. And these sensors are not made for exact measurements, these are manufactured to work in systems that need to alarm users when there is a high contamination of certain gasses, not to measure an actual value. These sensors have a limited life time and data drifts a lot over time. Also humidity and temperature have a big influence on the actual readings. 

You will notice if you combine these sensors you will get different readouts for the same gas over different sensors. Also these sensors need to be on for at least 15min before you can actually use them for reading data. And turning them on and off frequently will make them go bad pretty fast, because they heat up and cool down internally the actual sensor is oxidizing faster which will eventually degrade them to a point that getting a decent measurement out of them is hard.

Check out this [e-nose project](http://www.maskau.dk/projects/electronic-nose) from Michael Madsen.

**Comment by klynkins** [comment]
Thank you for the reply Cinezaster. I had a feeling they were going to give different readouts of the same gas on the sensors. We are looking for an accurate reading for carbon monoxide in PPM so I don't think we are going to continue using the MQ sensors. Thank you for the tips though.

Katelynn

**Comment by Cinezaster** [comment]
@klynkins 
There is a reason why you have > $100 sensors and the mq series which cost < $5.
You need to check datasheets for accuracy, resolution, overload(you can damage these sensors by putting them in a saturated environment) and cross sensitivity (if they measure different gasses, and by how much) 
These expensive sensors mitigate these issues by combining more sensors together and develop a algorithm that cancel out most of influencing effects. 
And they will calibrate each sensor before putting them on the market.

You might want to look at [these sensors for CO](http://www.winsen-sensor.com/products/co-module/). They are relatively cheap. Contain 3 electro chemical elements and they have internal temperature correction.
